src/main/groovy/de/felixschulze/gradle/HockeyAppUploadTask.groovy

```diff
@@ -102,23 +105,26 @@ class HockeyAppUploadTask extends DefaultTask {
             throw new IllegalArgumentException("Cannot upload to HockeyApp because API Token is missing")
         }
 
-        if (!applicationFile?.exists()) {
-            if (applicationFile) {
-                logger.debug("App file doesn't exist: "+applicationFile?.absolutePath)
-            }
+        if (!applicationFiles) {
             if (!applicationVariant && !hockeyApp.appFileNameRegex) {
                 throw new IllegalArgumentException("No appFileNameRegex provided.")
             }
             if (!hockeyApp.outputDirectory || !hockeyApp.outputDirectory.exists()) {
                 throw new IllegalArgumentException("The outputDirectory (" + hockeyApp.outputDirectory ? hockeyApp.outputDirectory.absolutePath : " not defined " + ") doesn't exists")
             }
-            applicationFile = FileHelper.getFile(hockeyApp.appFileNameRegex, hockeyApp.outputDirectory);
-            if (!applicationFile) {
+            applicationFiles = FileHelper.getFiles(hockeyApp.appFileNameRegex, hockeyApp.outputDirectory);
+            if (!applicationFiles) {
                 throw new IllegalStateException("No app file found in directory " + hockeyApp.outputDirectory.absolutePath)
             }
         }
 
-        logger.lifecycle("App file: " + applicationFile.absolutePath)
+        def appFilePaths = applicationFiles.collect { it.name }
+
+        if (!hockeyApp.allowMultipleAppFiles && applicationFiles.size() > 1) {
```

Comment:
It's a complicated issue. I'll try to answer in detail.
1. For most users this will not change anything because they probably have only one apk file in the applicationVariant.outputs. Even if they have more than one and somehow always get the one that should be (see below), after this change they just will get multiple APKs in HockeyApp instead of one. It's not what they were getting before (especially if it's important for them which one was uploaded last), but possibly it's still acceptable.
2. Previous version doesn't take the first apk. It takes SOME apk.
2.1. When using APK splits, the output files are in some random-looking order. I don't know whether there are other cases when it could be multiple output files or not and what is the order in these cases.
2.2. 'return true' at the end of the closure, passed to applicationVariant.outputs.each, is like 'continue' in the loop. For 'break' behavior it should be 'return false'.
2.3. By the way, FileHelper.getFile() also returns SOME matched file because File.list() gives unordered files. For this reason I think it would be right to throw an exception from this method if more than one file is matched. But such a change definitely could break something.
3. I think that taking not an exactly defined apk, but first, last or any apk - is not right anyway. The right way is to take on exactly defined or to take all of them.